### PR TITLE
Migrate to GitHub Actions

### PR DESCRIPTION
**Pull request by petergoldstein**
This PR sets up the repo with GitHub Actions and prawn-dev

It adds both tests and code-style as part of CI, and makes some linting changes to get code styling to pass.

Additional changes were necessary to get everything to green:

1. pdf-reader gem restrictions - something changed (not yet clear if deliberately or not) in this gem with version 2.9.0.  For the moment, exclude 2.9 and 2.9.1.  This issue is under investigation
2. Exclude 3.1, 3.1.0, and ruby-head from the CI matrix because of a transitive dependency on the matrix gem through prawn.  Like prawn-tables, this library needs an updated version of prawn to work properly on Ruby 3.1.0 and higher.

### CODE REVIEW

**Comment by petergoldstein** [comment]
An issue I filed in pdf-reader to discuss whether changing page_references return value was intentional - https://github.com/yob/pdf-reader/issues/443

**Comment by petergoldstein** [comment]
@yob has updated pdf-reader so that the issue will be fixed in the upcoming 2.9.2.  That means that the current `pdf-reader` restrictions will work, and that this PR is ready for review/merge.

@pointlessone can you please give this a look when you have a moment?  Thanks.

**Comment by petergoldstein** [comment]
@pointlessone This is rebased and green with the latest merge, so it is once again ready for review.

File: prawn-templates.gemspec [hunk 1]
```diff
@@ -20,13 +22,11 @@ Gem::Specification.new do |spec|
     'gregory.t.brown@gmail.com', 'brad@bradediger.com', 'dnelson@bluejade.com',
     'greenberg@entryway.net', 'jimmy@deefa.com', 'b.vogel@buddyandselly.com'
   ]
-  spec.add_dependency('pdf-reader', '~> 2.0')
+  spec.add_dependency('pdf-reader', '~> 2.0', '!= 2.9.0', '!= 2.9.1')
   spec.add_dependency('prawn', '~> 2.2')
   spec.add_development_dependency('pdf-inspector', '~> 1.3')
+  spec.add_development_dependency('prawn-dev', '~> 0.3.0')
   spec.add_development_dependency('rspec', '~> 3.0')
```
Comment (pointlessone) [inline prawn-templates.gemspec:29]: This is a dep of prawn-dev, too.

Comment (petergoldstein) [reply prawn-templates.gemspec]: @pointlessone the `rspec` line has been removed and this should be resolved.  Can you please take another look?  Thanks.

**Review by pointlessone — APPROVED** [review]

**Pull request merged**